### PR TITLE
Refactor templates into single template groups that allow you select TFM and aspire version

### PR DESCRIPTION
**Pull request by DamianEdwards**
## Description

This merges each template group into a single template that lets you select TFM and Aspire version.

Also addresses some outstanding content and dependency version issues.

Contributes to #6076

Some screenshots of this working in VS locally when I install the new template:

image

image

image

image

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue: dotnet/docs-aspire#1869
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6365)

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -71,7 +71,7 @@
     <NpgsqlEntityFrameworkCorePostgreSQLPackageVersion>8.0.8</NpgsqlEntityFrameworkCorePostgreSQLPackageVersion>
     <!-- for templates -->
     <MicrosoftExtensionsHttpResiliencePackageVersionForNet8>$(MicrosoftExtensionsHttpResiliencePackageVersion)</MicrosoftExtensionsHttpResiliencePackageVersionForNet8>
-    <MicrosoftExtensionsHttpResiliencePackageVersionForNet9>$(MicrosoftExtensionsHttpResiliencePackageVersion)</MicrosoftExtensionsHttpResiliencePackageVersionForNet9>
+    <MicrosoftExtensionsHttpResiliencePackageVersionForNet9>9.0.0-preview.9.24507.7</MicrosoftExtensionsHttpResiliencePackageVersionForNet9>
```
Comment (eerhardt) [inline eng/Versions.props:74]: My understanding is that `Microsoft.Extensions.Http.Resilience` is on a "tip support" policy. Meaning that only the latest version is supported. (I think there maybe a nuance here where the `8.0` version is LTS, but I can't seem to find the public policy.)

Given the above, I think we should always be referencing the current/latest version of `Microsoft.Extensions.Http.Resilience` from our templates regardless of TFM chosen. Even if you are targeting `net8.0`, we would bring in the 9.0 version of `Microsoft.Extensions.Http.Resilience`.

See also: [Use 8.0 era dependencies for non net9.0 TFMs (dotnet/extensions#5470)](https://github.com/dotnet/extensions/pull/5470)

cc @joperezr

File: src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json [hunk 2]
```diff
@@ -6,36 +6,88 @@
...
   "preferNameDirectory": true,
   "tags": {
     "language": "C#",
     "type": "project"
   },
   "precedence": "9000",
-  "identity": "Aspire.AppHost.CSharp.9.net9",
-  "thirdPartyNotices": "https://aka.ms/dotnet/aspire/8.0-third-party-notices",
-  "groupIdentity": "Aspire.AppHost.9",
+  "identity": "Aspire.AppHost.CSharp.8",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json:20]: Aspire.AppHost.CSharp.8.0

This needs to exactly match what is in the release/8.2 branch, or we will see duplicate options on the Additional Options page in Visual Studio on downlevel Visual Studio.

File: src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json [hunk 3]
```diff
@@ -6,36 +6,88 @@
...
     "Cloud"
   ],
-  "name": ".NET Aspire 9 App Host",
+  "name": ".NET Aspire App Host",
   "defaultName": "AppHost",
-  "description": "A project template for creating a .NET Aspire 9 app host (orchestrator) project.",
-  "shortName": ["aspire-apphost", "aspire-apphost-9"],
+  "description": "A project template for creating a .NET Aspire app host (orchestrator) project.",
+  "shortName": "aspire-apphost",
   "sourceName": "Aspire.AppHost1",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json:13]: Also, missing ide folder and icon for the AppHost template, as well as ide.host.json - note that you'll need to push in the AspireVersion symbol and mark as isVIsible as well.

File: src/Aspire.ProjectTemplates/templates/aspire-empty/.template.config/template.json [hunk 4]
```diff
@@ -10,21 +10,23 @@
...
     "language": "C#",
     "type": "solution",
-    "editorTreatAs": "solution"
+    "editorTreatAs": "solution",
+    "aspire-8.2-tfms": "net8.0",
+    "aspire-9.0-tfms": "net8.0;net9.0"
   },
   "precedence": "9000",
-  "identity": "Aspire.Empty.CSharp.9.net9",
+  "identity": "Aspire.Empty.CSharp.8",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-empty/.template.config/template.json:27]: 8.0 on the identity, also missing ide folder/icon and entries in ide.host.json

File: src/Aspire.ProjectTemplates/templates/aspire-mstest/.template.config/template.json [hunk 5]
```diff
@@ -11,33 +11,85 @@
...
   "tags": {
     "language": "C#",
     "type": "project"
   },
   "precedence": "9000",
-  "identity": "Aspire.Tests.MSTest.CSharp.9.net9",
-  "thirdPartyNotices": "https://aka.ms/dotnet/aspire/8.0-third-party-notices",
-  "groupIdentity": "Aspire.Tests.MSTest.9",
+  "identity": "Aspire.Tests.MSTest.CSharp.8",
+  "thirdPartyNotices": "https://aka.ms/dotnet/aspire/third-party-notices",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-mstest/.template.config/template.json:26]: 8.0 on the identity, also missing ide folder/icon and entries in ide.host.json

File: src/Aspire.ProjectTemplates/templates/aspire-nunit/.template.config/template.json [hunk 6]
```diff
@@ -11,33 +11,85 @@
...
   "sourceName": "Aspire.Tests.1",
   "preferNameDirectory": true,
   "tags": {
     "language": "C#",
     "type": "project"
   },
   "precedence": "9000",
-  "identity": "Aspire.Tests.NUnit.CSharp.9.net9",
-  "thirdPartyNotices": "https://aka.ms/dotnet/aspire/8.0-third-party-notices",
-  "groupIdentity": "Aspire.Tests.NUnit.9",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-nunit/.template.config/template.json:27]: 8.0 on the identity, also missing ide folder/icon and entries in ide.host.json

File: src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/.template.config/template.json [hunk 7]
```diff
@@ -10,33 +10,85 @@
...
   "preferNameDirectory": true,
   "tags": {
     "language": "C#",
     "type": "project"
   },
   "precedence": "9000",
-  "identity": "Aspire.ServiceDefaults.CSharp.9.net9",
-  "thirdPartyNotices": "https://aka.ms/dotnet/aspire/8.0-third-party-notices",
-  "groupIdentity": "Aspire.ServiceDefaults.9",
+  "identity": "Aspire.ServiceDefaults.CSharp.8",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/.template.config/template.json:24]: 8.0 on the identity, also missing ide folder/icon and entries in ide.host.json

File: src/Aspire.ProjectTemplates/templates/aspire-xunit/.template.config/template.json [hunk 8]
```diff
@@ -11,33 +11,85 @@
...
   "preferNameDirectory": true,
   "tags": {
     "language": "C#",
     "type": "project"
   },
   "precedence": "9000",
-  "identity": "Aspire.Tests.xUnit.CSharp.9.net9",
-  "thirdPartyNotices": "https://aka.ms/dotnet/aspire/8.0-third-party-notices",
-  "groupIdentity": "Aspire.Tests.xUnit.9",
+  "identity": "Aspire.Tests.xUnit.CSharp.8",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-xunit/.template.config/template.json:25]: 8.0 on the identity, also missing ide folder/icon and entries in ide.host.json

**Review by phenning — CHANGES_REQUESTED** [review]

**Review by phenning — COMMENTED** [review]
SO yea, ignore my comments about missing icons and ide.host.json files - you just need to add AspireVersion to the ide.host.jsons

File: eng/Versions.props (hunk 1 above)
Comment (DamianEdwards) [reply eng/Versions.props:74]: Shouldn't our repo be using the 9.x version then too? Currently it's using the 8.x version.

Comment (eerhardt) [reply eng/Versions.props:74]: Yes. That's part of this as well.

Comment (DamianEdwards) [reply eng/Versions.props:74]: Should I hold off on a change here until we get consensus?

File: src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json [hunk 9]
```diff
@@ -17,7 +17,7 @@
     "type": "project"
   },
   "precedence": "9000",
-  "identity": "Aspire.AppHost.CSharp.8",
+  "identity": "Aspire.AppHost.CSharp.8.8",
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-apphost/.template.config/template.json:20]: typo: 8.8

File: src/Aspire.ProjectTemplates/templates/aspire-empty/.template.config/ide.host.json [hunk 10]
```diff
@@ -3,6 +3,13 @@
   "icon": "ide/AspireEmpty.ico",
   "displayOverviewPage": "0",
   "disableHttpsSymbol": "NoHttps",
+  "symbolInfo": [
+    {
+      "id": "AspireVersion",
+      "isVisible": true,
+      "persistenceScope": "templateGroup"
```
Comment (phenning) [inline src/Aspire.ProjectTemplates/templates/aspire-empty/.template.config/ide.host.json:10]: So you perhaps want

"persistenceScope": "shared",
"persistenceScopeName": "aspireTemplates"

in order to sticky across all the aspire templates?

**Review by phenning — CHANGES_REQUESTED** [review]
one typo, one suggestion

File: eng/Versions.props (hunk 1 above)
Comment (joperezr) [reply eng/Versions.props:74]: I think there is an agreement here. As @eerhardt points out, we want to use the tip of this package both in net8 and net9. There is still the PR that he pointed out above which hasn't been merged but will be merged before 9.0 ships.

**Review by phenning — APPROVED** [review]
Good from an authoring metadata standpoint

**Pull request merged**